### PR TITLE
Add the default port documentation and remove incorrect CA note

### PR DESCRIPTION
**Pull request by brandisher**
Adding the default DNS-over-TLS port to the godoc and remove the note about multiple CAs as it's not applicable.

### CODE REVIEW

File: operator/v1/types_dns.go [hunk 1]
```diff
@@ -185,13 +186,13 @@ type DNSOverTLSConfig struct {
...
 	// caBundle references a ConfigMap that must contain either a single
-	// CA Certificate or a CA Bundle (in the case of multiple upstreams signed
-	// by different CAs). This allows cluster administrators to provide their
+	// CA Certificate or a CA Bundle. This allows cluster administrators to provide their
 	// own CA or CA bundle for validating the certificate of upstream resolvers.
 	//
 	// 1. The configmap must contain a `ca-bundle.crt` key.
 	// 2. The value must be a PEM encoded CA certificate or CA bundle.
 	// 3. The administrator must create this configmap in the openshift-config namespace.
+	// 4. The upstream server certificate must contain a Subject Alternative Name (SAN).
```
Comment (JoelSpeed) [inline operator/v1/types_dns.go:195]: A SAN or a specific SAN? Why does it need a SAN, any additional context we can add?

Comment (brandisher) [reply operator/v1/types_dns.go]: Excellent question. I fixed this in my latest push to note that the SAN must match `ServerName`.

**Review by soltysh — APPROVED** [review]
/approve

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Pull request merged**